aria_data_loaders/aria_data_utils/aria_live_streamer.py

```diff
@@ -0,0 +1,598 @@
+            - self.device_T_camera - Sophus SE3 transform from device frame (left-slam camera) to rgb-camera frame
+
+        Returns:
+            Tuple[aria.Device, aria.DeviceClient]: Tuple of aria's device and device client objects after their proper setup
+        """
+        # get a device client and configure it
+        device_client = aria.DeviceClient()
+        client_config = aria.DeviceClientConfig()
+        device_client.set_client_config(client_config)
+
+        # create a device
+        device = device_client.connect()
+
+        # get the path to this file and load the static device_T_cpf matrix
+        this_file_path = os.path.dirname(os.path.realpath(__file__))
+        self._device_T_cpf = np_matrix3x4_to_sophus_SE3(
+            np_matrix=np.load(
+                os.path.join(this_file_path, "../device_model/device_T_cpf.npy")
+            )
+        )
+
+        # create a streaming manager to manage subscriptions
+        streaming_manager = device.streaming_manager
+        streaming_client: aria.StreamingClient = streaming_manager.streaming_client  # type: ignore
+
+        # Get sensors calibration
+        self._sensors_calib_json = streaming_manager.sensors_calibration()
+        self._sensors_calib = device_calibration_from_json_string(
+            self._sensors_calib_json
+        )
+        self._rgb_calib_params = self._sensors_calib.get_camera_calib("camera-rgb")  # type: ignore
+        self._dst_calib_params = get_linear_camera_calibration(
+            512, 512, 280, "camera-rgb"
+        )
+        self.device_T_camera = sp.SE3(
+            self._sensors_calib.get_transform_device_sensor("camera-rgb").to_matrix()
+        )  # type: ignore
+
+        # Configure subscription to listen to Aria's RGB stream.
+        # streaming configuration
+        config = streaming_client.subscription_config
+        options = aria.StreamingSecurityOptions()
+        options.use_ephemeral_certs = False
+        options.local_certs_root_path = os.path.expanduser(
+            ("~/.aria/streaming-certs/persistent/")
+        )
+        config.security_options = options
+        config.subscriber_data_type = aria.StreamingDataType.Rgb
+        # A shorter queue size may be useful if the processing callback is always slow and you wish to process more recent data
+        # For visualizing the images, we only need the most recent frame so set the queue size to 1
+        # @TODO: Identify the best queue size for your application
+        config.message_queue_size[aria.StreamingDataType.Rgb] = 1  # type: ignore
+        config.message_queue_size[aria.StreamingDataType.Slam] = 1  # type: ignore
+        streaming_client.subscription_config = config
+
+        streaming_client.set_streaming_client_observer(self)
+
+        return device, device_client
+
+    def _rotate_img(self, img: np.ndarray, num_of_rotation: int = 3) -> np.ndarray:
```

Comment:
in general, a function like this can sit outside of a class, since it is not really specific to Aria. For example it doesn't use any of the objects, etc from the current class. This was it can be used elsewhere without needing to instantiate aria. Is there some version of utils.py that this can be moved to?

Reply:
I want us to create a 3rd common package within Spot-sim2real which houses all the shared perception functionalities+utils. That way Spot or Aria can just minimally use that as dependency. The alternative is housing this utils in Aria or Spot, where one now has a heavy-weight dependency on the other complex package.

Reply:
@akshararai @zephirefaith this should be a new PR.